### PR TITLE
Disable jquery migrate deprecation messages when prestashop debug mode is off

### LINKED ISSUE #15943: Jquery-migrate generates many messages in the console
Since upgrading jquery and adding jquery-migrate, we have many warning logs in the console:

**Screenshots**
Capture d’écran 2019-10-14 à 15 15 24

**Questions needing answers**

Should we remove them? Should it be kept here? Should we display it but only in dev environment ? If so, how is this done following best practices ?

### PR DESCRIPTION
**Pull request by matthieu-rolland**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | The console is flooded with jquery-migrate's deprecation messages, this PR disable those logs when the debug mode is off. It also adds a **isDebugMode** property accessible in javascript, in FO and in BO.
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #15943
| How to test?  | Display the home page in FO, and the dashboard page in BO: with debug mode ON, should have deprecation errors in the console as seen in the issue. When debug mode is OFF, they should not be there.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20738)

### CODE REVIEW

File: classes/controller/FrontController.php [hunk 1]
```diff
@@ -495,6 +495,7 @@ protected function assignGeneralPurposeVariables()
             'time' => time(),
             'static_token' => Tools::getToken(false),
             'token' => Tools::getToken(),
+            'isDebugMode' => _PS_MODE_DEV_,
```
Comment (eternoendless) [inline classes/controller/FrontController.php]: Use snake case for consistency.

Alternative variable name:
```suggestion
            'debug' => _PS_MODE_DEV_,
```

File: classes/controller/AdminController.php [hunk 2]
```diff
@@ -2724,6 +2725,8 @@ public function setMedia($isNewTheme = false)
             ]
         );
 
+        Media::addJsDef(['isDebugMode' => _PS_MODE_DEV_]);
```
Comment (eternoendless) [inline classes/controller/AdminController.php]: Maybe we could use this opportunity to introduce the `prestashop` variable in BO?
```suggestion
        Media::addJsDef([
            'prestashop' => [
                'debug' => _PS_MODE_DEV_,
            ],
        ]);
```

**Review by NeOMakinG — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by SD1982** [comment]
LGTM Thanks @matthieu-rolland !!

**Comment by Progi1984** [comment]
Thanks @matthieu-rolland

**Pull request merged**